src/backend/base/langflow/components/processing/regex.py

```diff
@@ -0,0 +1,91 @@
+import re
+
+from langflow.custom import Component
+from langflow.io import MessageTextInput, Output
+from langflow.schema import Data
+from langflow.schema.message import Message
+
+
+class RegexExtractorComponent(Component):
+    display_name = "Regex Extractor"
+    description = "Extract patterns from text using regular expressions."
+    icon = "regex"
+
+    inputs = [
+        MessageTextInput(
+            name="input_text",
+            display_name="Input Text",
+            info="The text to analyze",
+            required=True,
+        ),
+        MessageTextInput(
+            name="pattern",
+            display_name="Regex Pattern",
+            info="The regular expression pattern to match",
+            value=r"\b\w+@\w+\.\w+\b",  # Default pattern for email addresses
+            required=True,
+            tool_mode=True,
+        ),
+    ]
+
+    outputs = [
+        Output(display_name="Data", name="data", method="extract_matches"),
+        Output(display_name="Message", name="text", method="get_matches_text"),
+    ]
+
+    def extract_matches(self) -> list[Data]:
+        if not self.pattern or not self.input_text:
+            result = [Data(data={"matches": []})]
+            self.status = result
+            return result
+
+        try:
+            # Compile regex pattern
+            pattern = re.compile(self.pattern)
+
+            # Find all matches in the input text
+            matches = pattern.findall(self.input_text)
+
+            # Process matches and prepare result
+            if not matches:
+                result = [Data(data={"matches": []})]
+            else:
+                filtered_matches = [match for match in matches if match]  # Remove empty matches
+                if not filtered_matches:
+                    result = [Data(data={"matches": []})]
+                else:
+                    result = [Data(data={"match": match}) for match in filtered_matches]
```

Comment:
It's weird to have different result schemas.
Can we return an empty list if there's no match or only empty matches ?